source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,250 @@
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+The following sections feature examples that run text searches on the
+``fast_and_furious_movies`` collection. Each section uses a variable
+named "collection" to refer to the ``MongoCollection`` instance of the
+``fast_and_furious_movies`` collection. 
+
+The collection contains documents with a title and tags associated
+with one of the several movies that are part of the Fast and Furious
+movie franchise: 
+
+.. code-block:: json
+
+   { "_id": 1, "title": "2 Fast 2 Furious ", "tags": ["undercover", "drug dealer"] } 
+   { "_id": 2, "title": "Fast 5", "tags": ["bank robbery", "full team"] }
+   { "_id": 3, "title": "Furious 7", "tags": ["emotional"] }
+   { "_id": 4, "title": "The Fate of the Furious", "tags": ["betrayal"] }
+
+Text Index
+----------
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field on which to run a text
+search. A collection can only contain one text index.
+
+In the following examples, you run text searches on the ``title``
+field in the ``fast_and_furious_movies`` collection. To enable text
+searches on the ``title`` field, create a text index using the
+:ref:`Indexes <index-text-indexes>` builder with the following 
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+Multiple Indexes
+~~~~~~~~~~~~~~~~
+
+If you want to run text searches on both the ``title`` and
+``tags`` field in the ``fast_and_furious_movies`` collection, you
+need to create an text index to include them.
+
+You can create a :ref:`compound index <index-compound-indexes>`. A
+compound index is a single index that refers to multiple
+fields within a collection's documents. This allows you to specify
+multiple text fields using one text index.
+
+To enable text searches on the ``title`` and ``tag`` fields, create a
+compound index using the following snippet:  
+
+.. code-block:: java
+
+   collection.createIndex(Indexes.compoundIndex(Indexes.text("title"), Indexes.text("tags")));
+
+.. note::
+
+   If you have an existing text index on the collection, you need to drop
+   it. You can drop an index by calling the ``dropIndex()`` method,
+   specifying name of the text index as shown in the following snippet:
+
+   .. code-block:: java
+
+      collection.dropIndex("field_name");
+   
+   If you want to delete all existing indexes, specify "*" instead of
+   the name. Note that this does not delete the required index on the
+   ``_id`` field. 
+
+Text Search
+-----------
+
+Use the ``Filters.text()`` method to specify a text search. 
+
+The ``Filters.text()`` method uses the :doc:`Filters builder
+</fundamentals/builders/filters>` to create an instance of a :ref:`Bson
+<bson>` type. The instance is a query filter that specifies how the text
+search should operate, which you pass to the ``find()`` method. 
```

Comment:
Thought I had created a comment about this, but maybe I removed it by accident.
We should introduce what the Bson represents before the underlying type. A "Bson instance" could be anything you pass to MongoDB server, so we want to qualify it first. How about this:

```suggestion
The ``Filters.text()`` method uses the :doc:`Filters builder
</fundamentals/builders/filters>` to define a query filter based on
the search that you pass it. The query filter is represented by a 
Bson instance. Pass the query filter to the ``find()`` method to 
run your text search.
```